flyteadmin/pkg/manager/impl/validation/execution_validator.go

```diff
@@ -100,7 +100,13 @@ func CheckAndFetchInputsForExecution(
 			}
 			executionInputMap[name] = expectedInput.GetDefault()
 		} else {
-			inputType := validators.LiteralTypeForLiteral(executionInputMap[name])
+			var inputType *core.LiteralType
```

Comment:
nit: mind adding a unit test?

Reply:
yup planning to add more coverage. just needed reviews on the main flow

Reply:
done